File: app/src/main/java/com/tregula/graedukacyjna/domain/database/entity/Continent.java
package com.tregula.graedukacyjna.domain.database.entity;

import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "continents", indices = @Index(value = "name", unique = true))
public class Continent {

    @PrimaryKey(autoGenerate = true)
    public Long id;
    public final String name;
    public final Location location;
    public final String wiki;

    public Continent(String name, Location location, String wiki) {
        this.name = name;
        this.location = location;
        this.wiki = wiki;
    }
}

File: app/src/main/java/com/tregula/graedukacyjna/domain/database/entity/Continent.kt
package com.tregula.graedukacyjna.domain.database.entity

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "continents", indices = [Index(value = ["name"], unique = true)])
class Continent(val name: String,
                val location: Location,
                val wiki: String) {

    @PrimaryKey(autoGenerate = true)
    var id: Long? = null
}
